src/main/scala/com/cloudant/clouseau/IndexManagerService.scala

```diff
@@ -30,24 +30,18 @@ class IndexManagerService(ctx: ServiceContext[ConfigurationArgs]) extends Servic
 
   class LRU(initialCapacity: Int = 100, loadFactor: Float = 0.75f) {
 
-    class InnerLRU(initialCapacity: Int, loadFactor: Float) extends LinkedHashMap[String, Pid](initialCapacity, loadFactor, true) {
-
-      override def removeEldestEntry(eldest: Entry[String, Pid]): Boolean = {
-        val result = size() > ctx.args.config.getInt("clouseau.max_indexes_open", 100)
-        if (result) {
-          eldest.getValue ! ('close, 'lru)
-        }
-        result
-      }
-    }
+    class InnerLRU(initialCapacity: Int, loadFactor: Float) extends LinkedHashMap[String, Pid](initialCapacity, loadFactor, true)
 
+    val capacity = ctx.args.config.getInt("clouseau.max_indexes_open", 100)
     val lruMisses = metrics.counter("lru.misses")
     val lruEvictions = metrics.counter("lru.evictions")
 
     val pathToPid: JMap[String, Pid] = new InnerLRU(initialCapacity, loadFactor)
     val pidToPath: JMap[Pid, String] = new HashMap(initialCapacity, loadFactor)
 
     def get(path: String): Pid = {
+      assert(pathToPid.size == pidToPath.size)
+      enforceCapacity
```

Comment:
I wonder what the worst effect of the async-only `('close, 'lru)` could be. If the eldest entry takes a while to act on close message, we could pile a whole bunch `('close, 'lru)` messages into it's mailbox from every single get method call. Previously only `put` evicted entries.

On the other hand, with close being asynchronous, the LRU size could grow above the limit, so adding `enforceCapacity` to other methods would make sense. What if only `put`s and `remove`s enforce capacity. Each index on close, will remove itself and then `lru.remove` calls `enforceCapacity`, which might trigger more async closes and more `enforceCapacity` calls. Would that be enough?

Reply:
If we just do it in put (doing it remove seems pointless, as you'd expect) the map size goes past maximum and never comes back. doing it in get seems to keep things tight, I log the size of the lru and see good behaviour

however, perhaps even coupling it to lru functions is a mistake. by always returning `false` in removeEldestEntry (the default behaviour without the override) the maps are allowed to grow without bound. we need _something_ that will close when we're older and at sufficient pressure too. I don't think a 'send every' message pattern would suffice under pressure.

Reply:
Agree, we can't run `enforceCapacity` just in `put`. But maybe `put` + `remove` enough? It will end up with a chain of calls of `lru.remove` -> `enforceCapacity` -> close -> `lru.remove` -> `enforceCapacity` .... The chain should stop when it gets blow capacity. Maybe that's not fast enough?

Reply:
re-jigged this. put+remove is sufficient (frankly it only needs to be put, surely?), but because removal is async I send the close message to the eldest N entries, and this has the desired effect in local testing.

Reply:
removed the call in remove